src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -404,12 +411,106 @@ void FancyZones::MoveWindowIntoZone(HWND window, winrt::com_ptr<IZoneWindow> zon
     }
 }
 
+inline int RectWidth(const RECT& rect)
+{
+    return rect.right - rect.left;
+}
+
+inline int RectHeight(const RECT& rect)
+{
+    return rect.bottom - rect.top;
+}
+
+inline bool IsSameRect(const RECT& lhs, const RECT& rhs)
+{
+    return lhs.left   == rhs.left &&
+           lhs.top    == rhs.top &&
+           lhs.right  == rhs.right &&
+           lhs.bottom == rhs.bottom;
+}
+
+bool WindowInExpectedPosition(HWND window, const RECT& expected)
+{
+    WINDOWPLACEMENT placement{};
+    if (GetWindowPlacement(window, &placement))
+    {
+        return IsSameRect(expected, placement.rcNormalPosition);
+    }
+    return false;
+}
+
+RECT FitOnScreen(const RECT& windowRect, const RECT& originMonitorRect, const RECT& destMonitorRect)
+{
+    // New window position on active monitor. If window fits the screen, this will be final position.
+    int left = destMonitorRect.left + (windowRect.left - originMonitorRect.left);
+    int top = destMonitorRect.top + (windowRect.top - originMonitorRect.top);
+    int W = RectWidth(windowRect);
+    int H = RectHeight(windowRect);
+
+    if ((left < destMonitorRect.left) || (left + W > destMonitorRect.right))
+    {
+        // Set left window border to left border of screen (add padding). Resize window width if needed.
+        left = destMonitorRect.left + LEFT_TOP_PADDING;
+        W = min(W, RectWidth(destMonitorRect) - LEFT_TOP_PADDING);
+    }
+    if ((top < destMonitorRect.top) || (top + H > destMonitorRect.bottom))
+    {
+        // Set top window border to top border of screen (add padding). Resize window height if needed.
+        top = destMonitorRect.top + LEFT_TOP_PADDING;
+        H = min(H, RectHeight(destMonitorRect) - LEFT_TOP_PADDING);
+    }
+
+    return { left, top, left + W, top + H };

```

Comment:
For better readability
`return { .left = left, .top = top, .right = left + W, .bottom = top + H };`

Reply:
Nice improvement, done!